ui/lib/kv/addon/routes/list-directory.js

```diff
@@ -61,8 +62,18 @@ export default class KvSecretsListRoute extends Route {
 
   setupController(controller, resolvedModel) {
     super.setupController(controller, resolvedModel);
+    // listPermissions is used to show the overview card and/or an error message about LIST permissions.
+    resolvedModel.listPermissions = {
+      noDirectory: false,
+      noList: false,
+    };
+
     if (resolvedModel.secrets === 403) {
-      resolvedModel.noMetadataListPermissions = true;
+      resolvedModel.listPermissions.noList = true;
+      if (pathIsDirectory(resolvedModel.pathToSecret)) {
+        // the user attempted to return the list for a directory but does not have access
+        resolvedModel.listPermissions.noDirectory = true;
+      }
```

Comment:
I'm a little confused by this:
> // the user attempted to return the list for a directory but does not have access

Is the logic here that we get to this 403 if: 
1. a user queries a list they do not have permission for
2. a user queries a secret directory they do not have permission for
3. the secret directory does not exist